### PR TITLE
add support for Noah AVR PCB

### PR DESCRIPTION
**Pull request by kb-elmo**
## Description

Added support for my recently created AVR based replacement PCB for the MatrixLabs Noah.

The original PCB was based on an STM32 MCU that had no EEPROM so it wasn't possible to port it to VIA.
This one fixes that (plus some qol features like ESD protection).

https://github.com/kb-elmo/noah_avr

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kb_elmo/noah_avr/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# noah_avr
+
+![noah_avr](imgur.com image replace me!)
```
Comment (drashna) [inline keyboards/kb_elmo/noah_avr/readme.md:3]: Do you have an image to post here?

File: keyboards/kb_elmo/noah_avr/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [kb-elmo<mail@elmo.space>](https://github.com/yourusername)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
+
```
Comment (drashna) [inline keyboards/kb_elmo/noah_avr/readme.md:10]: You will want to fill this out.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kb_elmo/noah_avr/readme.md (hunk 2 above)
Comment (kb-elmo) [reply keyboards/kb_elmo/noah_avr/readme.md:10]: completely forgot the readme

File: keyboards/kb_elmo/noah_avr/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/kb_elmo/noah_avr/rules.mk:22]:
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = 65_ansi 65_iso
```

File: keyboards/kb_elmo/noah_avr/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,      KC_ENT,      KC_PGUP,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN,          
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_all(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      KC_TRNS,     KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
```
Comment (fauxpark) [inline keyboards/kb_elmo/noah_avr/keymaps/via/keymap.c:31]: Need two extra empty layers here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

File: keyboards/kb_elmo/noah_avr/noah_avr.h [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    k400, k401, k402,                   k406,                   k411, k412, k413, k414,       k415  \
+) { \
+    { k000, k001, k002, k003, k004, k005, k006, k007, k008, k009, k010, k011, k012, k013, k014, k015 }, \
+    { k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111, k112, k113, ____, k115 }, \
+    { k200, k201, k202, k203, k204, k205, k206, k207, k208, k209, k210, k211, ____, k213, ____, k215 }, \
+    { k300, k301, k302, k303, k304, k305, k306, k307, k308, k309, k310, k311, k312, k313, ____, k315 }, \
+    { k400, k401, k402, ____, ____, ____, k406, ____, ____, ____, ____, k411, k412, k413, k414, k415 }  \
+}
+
+#define LAYOUT_65_ansi( \
```
Comment (drashna) [inline keyboards/kb_elmo/noah_avr/noah_avr.h:45]: Looks like this doesn't match the `65_ansi` community layout.

Comment (tzarc) [reply keyboards/kb_elmo/noah_avr/noah_avr.h:45]: @drashna is this still an issue?

Comment (drashna) [reply keyboards/kb_elmo/noah_avr/noah_avr.h:45]: As long as it's added to `LAYOUTS`, then yeah.  The bottom row is missing a spot (it needs to have 6, not 5, like it has here)

Comment (drashna) [reply keyboards/kb_elmo/noah_avr/noah_avr.h:45]:
```suggestion
#define LAYOUT_64_ansi( \
```

File: keyboards/kb_elmo/noah_avr/noah_avr.h [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    k400, k401, k402,                   k406,                   k411, k412, k413, k414, k415  \
+) { \
+    { k000, k001, k002, k003, k004, k005, k006, k007, k008, k009, k010, k011, k012, k013, k014, k015 }, \
+    { k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111, k112, k113, ____, k115 }, \
+    { k200, k201, k202, k203, k204, k205, k206, k207, k208, k209, k210, k211, ____, k213, ____, k215 }, \
+    { k300, ____, k302, k303, k304, k305, k306, k307, k308, k309, k310, k311, k312, k313, ____, k315 }, \
+    { k400, k401, k402, ____, ____, ____, k406, ____, ____, ____, ____, k411, k412, k413, k414, k415 }  \
+}
+
+#define LAYOUT_65_iso( \
```
Comment (drashna) [inline keyboards/kb_elmo/noah_avr/noah_avr.h:73]:
```suggestion
#define LAYOUT_64_iso( \
```

File: keyboards/kb_elmo/noah_avr/noah_avr.h [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    k400, k401, k402,                   k406,                   k411, k412, k413, k414, k415  \
+) { \
+    { k000, k001, k002, k003, k004, k005, k006, k007, k008, k009, k010, k011, k012, k013, ____, k015 }, \
+    { k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111, k112, k113, ____, k115 }, \
+    { k200, k201, k202, k203, k204, k205, k206, k207, k208, k209, k210, k211, ____, k213, ____, k215 }, \
+    { k300, ____, k302, k303, k304, k305, k306, k307, k308, k309, k310, k311, k312, k313, ____, k315 }, \
+    { k400, k401, k402, ____, ____, ____, k406, ____, ____, ____, ____, k411, k412, k413, k414, k415 }  \
+}
+
+#define LAYOUT_65_ansi_splitbs( \
```
Comment (drashna) [inline keyboards/kb_elmo/noah_avr/noah_avr.h:59]:
```suggestion
#define LAYOUT_64_ansi_splitbs( \
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by tzarc — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**